### PR TITLE
responding to issue 2128

### PR DESCRIPTION
**Pull request by iamchrissmith**
reported in https://github.com/WP-API/WP-API/issues/2128 to check user against all visible post types

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@iamchrissmith it looks like there's a merge conflict here. Can you make sure it will merge cleanly?

File: lib/endpoints/class-wp-rest-users-controller.php [hunk 1]
```diff
@@ -188,6 +188,13 @@ public function get_item_permissions_check( $request ) {
 
 		$id = (int) $request['id'];
 		$user = get_userdata( $id );
+		$types = array();
+		foreach ( get_post_types( array(), 'object' ) as $obj ) {
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-users-controller.php]: Can't we just do `$types = get_post_types( array( 'show_in_rest' => true ), 'name' )`

**Comment by iamchrissmith** [comment]
@danielbachhuber yes, that works a lot better.  Merge conflicts also resolved (I hope)

**Comment by danielbachhuber** [comment]
@iamchrissmith will you be adding a test for this?

**Comment by iamchrissmith** [comment]
I added a test for pages copied from the `test_get_item_published_author` and changed the name of that one to `test_get_item_published_author_post` to clarify.  I ran out of time trying to get PHPUnit setup locally to test the test, but wanted to get these up in case someone has time to test the tests

**Comment by danielbachhuber** [comment]
Thanks for your work on this, @iamchrissmith! I've started a new PR with your branch, and fixed up the tests: #2155

**Pull request closed without merging**